### PR TITLE
add multiple os to matrix strategy

### PR DESCRIPTION
**Pull request by kaitejohnson**
Attempting to test on ubuntu, windows, and mac...

### CODE REVIEW

**Comment by seabbs** [comment]
> Run Rscript -e 'cmdstanr::check_cmdstan_toolchain(fix = TRUE)'
> Installing mingw32-make with Rtools40
> The C++ toolchain required for CmdStan is setup properly!
> Run Rscript -e 'cmdstanr::install_cmdstan(version = "2.35.0", cores = 2)'
> Error: Error: unexpected numeric constant in "cmdstanr::install_cmdstan(version = 2.35.0"
> Execution halted
> Error: Process completed with exit code 1.

The remaining issue is with R 4.1 and is very odd. It looks to be due to the way that windows on that version of R only is interpreting strings?

**Comment by seabbs** [comment]
Note this is currently pinned to a fixed release but needs to be changed back to the rolling release once that has been updated

**Comment by seabbs** [comment]
@kaitejohnson I think cmdstanr wise this is now working and is "stable". The mac dependency installation issue is a odd one. I haven't seen the package that is causing the issue before.

File: .github/workflows/r-cmd-check.yaml [hunk 1]
```diff
@@ -8,9 +8,11 @@ on:
 jobs:
   check-package:
     strategy:
+      fail-fast: false
```
Comment (seabbs) [inline .github/workflows/r-cmd-check.yaml:11]: you might want to revert this as it makes the CI less wasteful (the whole matrix fails if one entry fails).

Comment (kaitejohnson) [reply .github/workflows/r-cmd-check.yaml:11]: I probably copied this blindly from somewhere.

**Comment by kaitejohnson** [comment]
Can you describe briefly what you changed here @seabbs ?

File: .github/workflows/r-cmd-check.yaml (hunk 1 above)
Comment (seabbs) [reply .github/workflows/r-cmd-check.yaml]: No I added it for testing so I could see what worked and what didn't.

Comment (seabbs) [reply .github/workflows/r-cmd-check.yaml]: I prefer to have it on to avoid waste but you do get more information if its off.

**Comment by seabbs** [comment]
I updated the rolling release of `install-cmdstan` to use the `gh` cli with a GITHUB secret rather than using the API directly with a API secret. I also switched to using javascript rather than custom bash and powershell across OS's to increase stability. 

The last change I needed to make in order to get this working with R 4.1 and windows (i.e it already worked on the latest release) was to be careful about the use of " and ' quotes. Apparently windows sometimes treated these interchangeable (but only when used in one order). 

Its not relevant here but I also added support for installing `cmdstanr` in the action if its not present in the calling action. I tried to do this in R but couldn't get it to only install R if not present vs all the time (mostly due actually neededing to try and detect the user having called `setup-r` in the wrapper step vs R (as the built in R can't be used to install packages).

The code changes here were almost entirely exploratory and should now be mostly set back to what you had. I found this still failed on mac but that this failure was unrelated to cmdstan (its something in your dependencies that I have never seen before).

**Comment by kaitejohnson** [comment]
Wow thank you @seabbs. I'll pretend I understood 50% of that. 

I think I am in favor of the rolling release so we internally know when breaking changes are introduced (if I am understanding correctly), even 

Looks like the ruleset names got updated because these all ran.

**Review by dylanhmorris — APPROVED** [review]
This LGTM, and I will update the branch protection rules. Only one question: is there a reason not to add macOS now as well, @seabbs / @kaitejohnson?

**Comment by seabbs** [comment]
Note also that all windows VM seem very slow at downloading cache objects which makes them run slower than they otherwise would. This is new behaviour as far as I am aware and so will hopefully be transient.

**Comment by kaitejohnson** [comment]
@dylanhmorris I now can't seem to change the names of the checks anywhere -- I thought it was in the ruleset but it is not an option I am seeing...

**Comment by dylanhmorris** [comment]
> @dylanhmorris I now can't seem to change the names of the checks anywhere -- I thought it was in the ruleset but it is not an option I am seeing...

I already did that, but for future reference:

`Settings > Rules > Rulesets > (click on the ruleset to modify)`

then scroll to "Branch rules"; find "Require status checks to pass", and click on "Show additional settings'

**Pull request merged**